subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/daemon/CompilerDaemonClient.java

```diff
@@ -32,7 +35,12 @@ public CompilerDaemonClient(DaemonForkOptions forkOptions, CompilerDaemonWorker
     public <T extends CompileSpec> CompileResult execute(Compiler<T> compiler, T spec) {
         // currently we just allow a single compilation thread at a time (per compiler daemon)
         // one problem to solve when allowing multiple threads is how to deal with memory requirements specified by compile tasks
-        return workerProcess.execute(compiler, spec);
+        BuildOperationWorkerRegistry.Completion workerLease = buildOperationWorkerRegistry.operationStart();
```

Comment:
Per our conversation, this is probably fine for now.  We may want to discuss again later when we tackle memory management.